<launch>
  <arg name="camera" value="foscam"/>
  <arg name="robot_tag" value="scarab"/>
  <arg name="robot_base" value="$(arg robot_tag)/odom_laser"/>
  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.01, frame_id: scarab},
      {id: 1, size: 0.163, frame_id: tag_1},
      {id: 2, size: 0.163},
      {id: 3, size: 0.163}
     ] 
    </rosparam>
    <param name="sensor_frame_id" value="$(arg camera)"/>
    
    <remap from="/camera_info" to="/$(arg camera)/camera_info"/>
    <remap from="/image_rect" to="/$(arg camera)/image_rect"/>
  </node>

  <node pkg="apriltags_ros" type="apriltag_map_localizer" name="apriltag_map_localizer" output="screen">
    <param name="camera" value="$(arg camera)"/>
    <param name="robot_tag" value="$(arg robot_tag)"/>
    <param name="robot_base" value="$(arg robot_base)"/>
  </node>
</launch>
